src/org/zaproxy/zap/extension/ascanrulesAlpha/HttpOnlySite.java

```diff
@@ -0,0 +1,156 @@
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import org.apache.commons.httpclient.URIException;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.core.scanner.AbstractHostPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpStatusCode;
+import org.parosproxy.paros.Constant;
+
+import java.io.IOException;
+
+/**
+ * Active scan rule which raises an alert if a site accessed via HTTP is also available under HTTPS 
+ * https://github.com/zaproxy/zaproxy/issues/2207
+ * @author sanchitlucknow@gmail.com
+ */
+public class HttpOnlySite extends AbstractHostPlugin {
+
+	/**
+	 * Prefix for internationalised messages used by this rule
+	 */
+	private static final String MESSAGE_PREFIX = "ascanalpha.httponlysite.";
+	private static final int PLUGIN_ID = 10147;
+
+	private static final Logger log = Logger.getLogger(HttpOnlySite.class);
+	
+	@Override
+	public int getId() {
+		return PLUGIN_ID;
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "name");
+	}
+	
+	public String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return null;
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.MISC;
+	}
+	
+	@Override
+	public int getRisk() {
+		return Alert.RISK_LOW;
+	}
+
+	@Override
+	public int getCweId() {
+		return 3111; // CWE-311: Missing Encryption of Sensitive Data
+	}
+
+	@Override
+	public int getWascId() {
+		return 43; // WASC-04: Insufficient Transport Layer Protection
+	}
+
+	@Override
+	public void init() {
+
+	}
+
+	@Override
+	public void scan() {
+
+		if (this.isStop()) { // Check if the user stopped things
+			if (log.isDebugEnabled()) {
+				log.debug("Scanner "+this.getName()+" Stopping.");
+			}
+			return; // Stop!
+		}
+		
+		if (this.getBaseMsg().getRequestHeader().getSecure()) { //Base request is HTTPS
+			if (log.isDebugEnabled())  log.debug ("The original request was HTTPS, so there is not much point in looking further.");
+			System.out.println("\n\nBase isnt HTTP\n\n");
+			return;	//Get out
+		}
+
+		HttpMessage newRequest = this.getNewMsg();
+		System.out.println(newRequest.getRequestHeader().toString());
+		try{
+			newRequest.getRequestHeader().setSecure(true); //https becomes http
+			log.info("**"+newRequest.getRequestHeader().getURI());
+			System.out.println(newRequest); //Checking the request
+		} catch (URIException e) {
+			log.error("Error creating HTTPS URL from HTTP URL:", e);
+			return; //Get out
+		}
+
+		try {
+			sendAndReceive(newRequest, true);
```

Comment:
It should not follow redirections automatically, otherwise you might get a response that does not correspond to the original request.
For example, the requests to https://example.com always redirect to http://example.com, in this case it would not raise an alert but the server is configured to always use HTTP over HTTPS.